diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +109,191 @@ class LineLength : Rule("line-length") {
         }
     }
 
+    private fun fixError(wrongNode: ASTNode, configuration: LineLengthConfiguration, grand: ASTNode) {
+        if (wrongNode.elementType == EOL_COMMENT) {
+            fixComment(wrongNode, configuration)
+            return
+        }
+        var parent = wrongNode
+        do {
+            when (parent.elementType) {
+                FUN -> {
+                    if (!parent.hasChildOfType(BLOCK))
+                        parent.addChild(PsiWhiteSpaceImpl("\n"), parent.findChildByType(EQ)!!.treeNext)
+                    return
+                }
+                CONDITION -> {
+                    fixCondition(parent, configuration, grand)
+                    return
+                }
+                PROPERTY -> {
+                    fixProperty(parent, grand, configuration)
+                    return
+                }
+                else -> parent = parent.treeParent
+            }
+        } while (parent.treeParent != null)
+    }
+
+    private fun fixComment(wrongNode: ASTNode, configuration: LineLengthConfiguration){
+        var node = wrongNode
+        do {
+            val commentText = node.text
+            var indexLastSpace = configuration.lineLength.toInt()
+            while (commentText[indexLastSpace] != ' ') {
+                --indexLastSpace
+                if (indexLastSpace == 0)
+                    return
+            }
+            val nodeText = "//${commentText.substring(indexLastSpace, commentText.length)}"
+            val newNode = LeafPsiElement(EOL_COMMENT, nodeText)
+            node.treeParent.addChild(LeafPsiElement(EOL_COMMENT, commentText.substring(0, indexLastSpace)), node)
+            node.treeParent.addChild(PsiWhiteSpaceImpl("\n"), node)
+            node.treeParent.addChild(newNode, node)
+            node.treeParent.removeChild(node)
+            node = newNode
+        } while (node.text.length <= configuration.lineLength)
+    }
+
+    private fun fixCondition(wrongNode: ASTNode, configuration: LineLengthConfiguration, grand: ASTNode,
+                             leftInitOffset: Int = -1, isProperty: Boolean = false) {
+        var leftOffset = if (leftInitOffset < 0){
+            val positionByOffset = calculateLineColByOffset(grand.treeParent.text)
```

Comment:
If I understand correctly, you always call `calculateLineColByOffset` with `grand.treeParent` as argument, and `grand.treeParent` is the root node. So it seems that you can calculate it once when entering `visit` function.